scripts/open-coredump.sh

```diff
@@ -0,0 +1,291 @@
+#!/bin/bash
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022-present ScyllaDB
+#
+
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+#
+# Requirements:
+# * eu-unstrip
+# * curl
+# * jq
+# * git
+
+set -e
+
+function print_usage {
+cat << EOF
+Usage: ./open-coredump.sh [options] COREFILE
+
+Script for "one-click" opening of coredumps.
+
+It extracts the build-id from the coredump, retrieves metadata for that
+build, downloads the binary package, the source code and finally
+launches the dbuild container, with everything ready to load the
+coredump.
+
+The script is idempotent: running it after the prepartory steps will
+re-use what is already donwloaded. It is *strongly* recommended to run
+this from an empty directory, with nothing but the core-file present.
+
+Options:
+
+--help,-h
+    Print this help message and exit.
+
+--quiet,-q
+    Only print errors (if any).
+
+--progress
+    Print progress information when downloading the package and cloning
+    the git repo.
+
+--artifact-dir,-d ARTIFACT_DIR
+    Directory where the scrip will store all downloaded artifacts. If
+    the directory doesn't exist, it will be created.
+    It is strongly recommended that this is an empty directory, not used
+    for anything else, to avoid interference with the script.
+    To reset the state of the script, just delete this directory.
+    This can be the same as the directory the coredump is already in, or
+    some other path.
+    By default, a directory called open-coredump.sh.dir will be created in
```

Comment:
nit: you can use `basename $0` instead of `open-coredump.sh` so in case the script is renamed
the help message will reflect that.

Reply:
Good idea. Initially I used `$0` but that included the full path and I didn't think of using `basename` on it.

Reply:
Done in v4.